### DISCUSSION TITLE
Web Search (with SearXNG) Stopped Working

### DISCUSSION THREAD

**Discussion by misterjice** [open-webui/open-webui#7962] [Issues]
# Bug Report

---

## Installation Method

Installed with pip

## Environment

- **Open WebUI Version:**  v0.4.8
- **Ollama Version:** 0.5.3

- **Operating System:** Windows 11
- **Browser):** Firefox

**Confirmation:**

## Expected Behavior:

I expect to get search results.  

## Actual Behavior:

I get an error.

## Description

**Bug Summary:**
Web search is no longer working and my AI is telling me this has something to do with the OpenWebUI scripts, specifically:
- ERROR [open_webui.apps.retrieval.main] 'NoneType' object has no attribute 'encode'
- line 854, in save_docs_to_vector_db    embeddings = embedding_function(
- line 248, in <lambda> return lambda query: embedding_function.encode(query).tolist()
- line 1330, in process_web_search    save_docs_to_vector_db(docs, collection_name, overwrite=True)
- line 876, in save_docs_to_vector_db raise e
- line 854, in save_docs_to_vector_db     embeddings = embedding_function(
- ine 248, in <lambda>   return lambda query: embedding_function.encode(query).tolist()

## Reproduction Details

**Steps to Reproduce:**
I ask it to do something that requires a search query.  I toggle the Web Search.  I wait.  I get an error instead of search results.  

## Logs and Screenshots

**Browser Console Logs:**
Fetching pages: 100%|##################################################################| 12/12 [00:00<00:00, 16.98it/s]
INFO  [open_webui.apps.retrieval.main] save_docs_to_vector_db: document Ukraine War | Latest News & Updates| BBC News, Ukraine War Situation Update: 12 – 18 October 2024, Browser check - Consilium, Ukraine Conflict Updates | Institute for the Study of War, Ukraine: Zelenskyy wants 'strong' Trump 'on our side' – DW – 12/19/2024, Putin says he’s ‘ready’ for potential talks with Trump during year-end news conference | CNN, Vladimir Putin claims war in Ukraine has made Russia ‘much stronger’ – The Irish Times, Russia-Ukraine updates: Zelenskyy slams Putin’s missile ‘duel’ challenge | Russia-Ukraine war News | Al Jazeera, reuters.com, Putin boasts of gains in Ukraine at annual news conference, Russia-Ukraine war: List of key events, day 1,029 | Russia-Ukraine war News | Al Jazeera 7ff08330faedbec27351d14696259393e3f84ed33d189a2259b05462c0a19c7
INFO  [open_webui.apps.retrieval.main] adding to collection 7ff08330faedbec27351d14696259393e3f84ed33d189a2259b05462c0a19c7
ERROR [open_webui.apps.retrieval.main] 'NoneType' object has no attribute 'encode'
Traceback (most recent call last):
  File "C:\Users\J\artroom\miniconda3\envs\OpenWebUI\Lib\site-packages\open_webui\apps\retrieval\main.py", line 854, in save_docs_to_vector_db
    embeddings = embedding_function(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\J\artroom\miniconda3\envs\OpenWebUI\Lib\site-packages\open_webui\apps\retrieval\utils.py", line 248, in <lambda>
    return lambda query: embedding_function.encode(query).tolist()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'encode'
ERROR [open_webui.apps.retrieval.main] 'NoneType' object has no attribute 'encode'
Traceback (most recent call last):
  File "C:\Users\J\artroom\miniconda3\envs\OpenWebUI\Lib\site-packages\open_webui\apps\retrieval\main.py", line 1330, in process_web_search
    save_docs_to_vector_db(docs, collection_name, overwrite=True)
  File "C:\Users\J\artroom\miniconda3\envs\OpenWebUI\Lib\site-packages\open_webui\apps\retrieval\main.py", line 876, in save_docs_to_vector_db
    raise e
  File "C:\Users\J\artroom\miniconda3\envs\OpenWebUI\Lib\site-packages\open_webui\apps\retrieval\main.py", line 854, in save_docs_to_vector_db
    embeddings = embedding_function(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\J\artroom\miniconda3\envs\OpenWebUI\Lib\site-packages\open_webui\apps\retrieval\utils.py", line 248, in <lambda>
    return lambda query: embedding_function.encode(query).tolist()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'encode'
INFO:     127.0.0.1:65277 - "POST /retrieval/api/v1/process/web/search HTTP/1.1" 400 Bad Request
INFO  [open_webui.apps.openai.main] get_all_models()
INFO  [open_webui.apps.ollama.main] get_all_models()
INFO  [open_webui.apps.ollama.main] url: http://localhost:11434
INFO:     127.0.0.1:65277 - "POST /ollama/api/chat HTTP/1.1" 200 OK

**Comment by tjbck** [comment]
Definitely peculiar, can you reproduce the issue with the dev branch?

**Reply by misterjice** [reply]
I figured out where dev branch is.  I haven't installed it though.  And issue is gone with update.

**Comment by moofya** [comment]
same question in main branch
![5C833E6F-4B25-40bb-85FA-03689856799A](https://github.com/user-attachments/assets/2d5e3c05-6d1b-4346-a93e-0eec1a052ed4)
